scout/config.py

```diff
@@ -0,0 +1,150 @@
+        self.set_config_args(self.cli_args)
+
+    @classmethod
+    def _load_config(cls, filepath):
+        with open(filepath, "r") as file:
+            return yaml.safe_load(file)
+
+    def _validate(self, input_data: dict, schema_data: dict = None):
+        """Validate argument data against schema data
+
+        Args:
+            input_data (dict): Data to validate.
+            schema_data (dict, optional): Data to validate against, "None" value will use the yml
+                                          schema data in supporting_data/config_schema.
+        """
+
+        if not schema_data:
+            schema_data = self.schema_data
+        validate(input_data, schema_data)
+
+    def initialize_argparse(self, parser):
+        # Initialize arguments with a yml config file argument
+        parser.add_argument(
+            "-y", "--yaml",
+            type=Path,
+            help=("Path to YAML configuration file, arguments in this file will take priority over "
+                  "arguments passed via the CLI")
+            )
+
+    def set_config_args(self, cli_args: list[str] = []):
+        # If applicable, set yml arguments to dict object
+        self.config_args = {}
+        self.args = self.parser.parse_args(cli_args)
+        if self.args.yaml:
+            self.config_args = self._load_config(self.args.yaml)
+            self._validate(self.config_args, self.schema_data)
+
+    def parse_args(self):
+        # Update args with yaml arguments
+        if self.config_args:
+            config_key_args = self.config_args.get(self.key, {})
+            self.args = self.update_args(self.args, config_key_args)
+
+        # Ensure command-line args take precendence
+        self.args = self.parser.parse_args(self.cli_args, namespace=self.args)
+
+        # Check for valid arguments
+        self.check_dependencies(self.args)
+
+        return self.args
+
+    def update_args(self, existing_args: argparse.NameSpace, new_args: dict):  # noqa: F821
+        """ Update argparse arguments NameSpace with dictionary of args"""
+
+        for k, v in new_args.items():
+            if isinstance(v, dict):
+                self.update_args(existing_args, v)
+            else:
+                existing_args.__dict__.update({k: v})
+
+        return existing_args
+
+    def check_dependencies(self, args: argparse.NameSpace):  # noqa: F821
+        # Verify that arguments are valid that are not captured in the yml schema
+        if "fuel types" in vars(args).get("detail_brkout", []) and args.split_fuel is True:
+            raise ValueError(
+                "Detailed breakout (detail_brkout) cannot include `fuel types` if split_fuel==True")
+
+    def create_argparse(self, parser, schema_data, group=None):
+        """Extracts arguments from the config schema and writes argparse arguments"""
+        if group:
+            parser = group
+        arguments = copy.deepcopy(schema_data).get("properties", {})
```

Comment:
Could you add a handful of comments to this function to help explain what the different parts are doing?